sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -146,12 +120,28 @@ public static BinaryData fromStream(InputStream inputStream) {
      * @return A {@link Mono} of {@link BinaryData} representing the {@link Flux} of {@link ByteBuffer}.
      */
     public static Mono<BinaryData> fromFlux(Flux<ByteBuffer> data) {
-        if (Objects.isNull(data)) {
-            return Mono.just(EMPTY_DATA);
-        }
+        return fromFlux(data, null);
+    }
 
-        return FluxUtil.collectBytesInByteBufferStream(data)
-            .flatMap(bytes -> Mono.just(new BinaryData(bytes)));
+    /**
+     * Creates an instance of {@link BinaryData} from the given {@link Flux} of {@link ByteBuffer}.
+     * <p>
+     * If the {@code data} is null an empty {@link BinaryData} will be returned.
+     *
+     * <p><strong>Create an instance from a Flux of ByteBuffer</strong></p>
+     *
+     * {@codesnippet com.azure.core.util.BinaryData.fromFlux#Flux}
+     *
+     * @param data The {@link Flux} of {@link ByteBuffer} that {@link BinaryData} will represent.
+     * @param length The length of {@code data} in bytes.
+     * @return A {@link Mono} of {@link BinaryData} representing the {@link Flux} of {@link ByteBuffer}.
+     */
+    public static Mono<BinaryData> fromFlux(Flux<ByteBuffer> data, Long length) {
```

Comment:
Can this be little-L `long`?

Reply:
I agree, this should be `long` instead of `Long`. There will need to be some code path duplication to support this as it appears `Long` was chosen to allow for code de-duplication.